### PR TITLE
Number of fixes for gyp builds

### PR DESCRIPTION
**Pull request by hintjens**
Also: raised socket limit to 16K on Windows in CMakeLists.txt

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -434,7 +422,7 @@ AS_IF([test "x$with_libsodium" = "xyes"], [
 
 AC_ARG_ENABLE([curve],
     AS_HELP_STRING([--disable-curve], [disable CURVE security [default=no]]))
-
+x
```
Comment (bluca) [inline configure.ac]: What's this X? :-)

Comment (hintjens) [inline configure.ac]: Huh.

Fixed, commit sent.

Comment (hintjens) [inline configure.ac]: Oh, and thanks for catching that!

**Comment by bluca** [comment]
:+1: especially on the pedantic warnings, it's good to have that enabled by default.

**Pull request merged**